packages/ntcore-ts-client/src/lib/pubsub/topic.ts

```diff
@@ -13,7 +13,7 @@ import type {
   TopicProperties,
 } from '../types/types';
 
-export class NetworkTablesTopic<T extends NetworkTablesTypes> {
+export class NetworkTablesTopic<T extends NetworkTablesTypes = any> {
```

Comment:
nit: I would like clarification to changes in this file

Reply:
So a lot of my changes required creating or passing around NetworkTablesTopics without knowing what the generic type would be. 
You got around this before by directly setting it to `<any>`:

But that also required you to disable the eslint rule:
```typescript
// eslint-disable-next-line @typescript-eslint/no-explicit-any
private topics: Map<string, NetworkTablesTopic<any>>;
```
I wanted to avoid adding that eslint comment all over the place.

Since the users of the library doesn't ever directly create a NetworkTablesTopic, I thought it would be cleaner to be able to refer to NetworkTablesTopics without a generic type. 
`any` was my first attempt at that, but I just played around with this instead:
```typescript
export class NetworkTablesTopic<T extends NetworkTablesTypes = NetworkTablesTypes> {
```
Which I do like a lot better, but required two spots to cast to `unkown` first, which I don't love:
```typescript
  constructor(client: PubSubClient, name: string, typeInfo: NetworkTablesTypeInfo, defaultValue?: T) {
    this.client = client;
    this._name = name;
    this._typeInfo = typeInfo;
    this.value = defaultValue ?? null;
    this._announced = false;
    this._publisher = false;
    this._subscribers = new Map();

    const existingTopic = this.client.getTopicFromName(name);
    if (existingTopic) {
      if (existingTopic.typeInfo[0] === typeInfo[0] && existingTopic.typeInfo[1] === typeInfo[1]) {
        return existingTopic as unknown as NetworkTablesTopic<T>;
      } else {
        throw new Error(`Topic ${name} already exists, but with a different type.`);
      }
    }

    this.client.registerTopic(this as unknown as NetworkTablesTopic);
  }
```

Let me know if you a better TS workaround!